Docker image no longer builds

The issue here is that the version of cargo available for alpine always lags a bit behind. As soon as the repo uses non supported features, the build will break until a newer version of cargo for alpine makes it in.

I think we may want to reconsider using alpine for now at least as default and bring back the phusion image as default. Once substrate stabilises, there is no doubt the alpine version will be much better but for now, I see mostly extra hassle.

cc @azban, your thoughts?

```
 Compiling twox-hash v1.1.1
error[E0658]: :vis fragment specifier is experimental and subject to change (see issue #41022)
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/fixed-hash-0.3.0-beta.3/src/hash.rs:51:22
   |
51 |     ( $(#[$attr:meta])* $visibility:vis struct $name:ident ( $n_bytes:expr ); ) => {
   |                         ^^^^^^^^^^^^^^^

error[E0432]: unresolved import `static_assertions::const_assert`
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/fixed-hash-0.3.0-beta.3/src/lib.rs:27:9
   |
27 | pub use static_assertions::const_assert;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `const_assert` in the root. Did you mean to use `const_assert_eq`?

error: aborting due to 2 previous errors

Some errors occurred: E0432, E0658.
For more information about an error, try `rustc --explain E0432`.
error: Could not compile `fixed-hash`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,12 +1,12 @@
-FROM frolvlad/alpine-glibc AS builder
+FROM alpine:edge AS builder
 LABEL maintainer="chevdor@gmail.com"
 LABEL description="This is the build stage for Substrate. Here we create the binary."
 
 RUN apk add build-base \
     cmake \
     linux-headers \
-    openssl-dev && \
-    apk add --repository http://nl.alpinelinux.org/alpine/edge/community cargo
+    openssl-dev \
+    cargo
 
 ARG PROFILE=release
 WORKDIR /substrate

